product_packaging_dimension/models/product_packaging.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class ProductPackaging(models.Model):
+    _inherit = "product.packaging"
+
+    max_weight = fields.Float()
+    length = fields.Integer('Length (mm)', help='length in millimeters')
+    width = fields.Integer('Width (mm)', help='width in millimeters')
+    height = fields.Integer('Height (mm)', help='height in millimeters')
+    volume = fields.Float(
+        'Volume (m³)',
+        digits=(8, 4),
+        compute='_compute_volume',
+        readonly=True,
+        store=False,
+        help='volume in cubic meters',
+    )
+
+    @api.depends('length', 'width', 'height')
+    def _compute_volume(self):
+        for pack in self:
```

Comment:
add a warning this compute work only if we stay with `'length', 'width', 'height'` in mm :wink: 

and I will now why we use length ... in mm and volumen not in mm ?

Reply:
Following https://en.wikipedia.org/wiki/International_System_of_Units, I would say meters for length and m3 for volume.
--

Reply:
The issue in using meters is that we'd need a Float field type as meter is too big of a uom for parcel dimensions.
However odoo already defines these fields as Integer in `delivery` !
As there is no functional dependency between this module and `delivery` we must use the same field type in order to avoid conflicts...

Reply:
What are the drawbacks using float with at least 4 decimals ?

Reply:
I think it's better to not enforce any UoM because it depends too much of the business cases or even locales (non-metric systems). Adding a length_uom_id can solve this debate. 


Please also have a look on this PR https://github.com/odoo/odoo/pull/50252 to make these uom_id first class citizen.

Reply:
@hparfr It's been a while I had read your comments and forgot about it until I got back to having to handle uom for weight and volume. 
To me, your PR on odoo looks like the right direction, and we should probably update our modules to consider at least the `ir.config_parameter` instead of having uom in string descriptions.